recipes/msgpack-cxx/all/conanfile.py

```diff
@@ -85,4 +90,5 @@ def package_info(self):
         self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
         self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
 
-        self.cpp_info.defines = ["MSGPACK_USE_BOOST"]
+        if tools.Version(self.version) >= "4.1.0" and not self.options.use_boost:
```

Comment:
```suggestion
        if not self.options.get_safe("use_boost", True):
```
In case removed or true, it will not pass.

Reply:
No, same comment than https://github.com/conan-io/conan-center-index/pull/9600#discussion_r826888908

Reply:
it would be quite obfuscated, but valid.